frontend/src/StartView.svelte

```diff
@@ -134,15 +136,33 @@
       showRootResource = false;
       return;
     }
-
-    const rootModel = await fetch(
-      `${$settings.backendUrl}/create_root_resource?name=${f.name}`,
-      {
-        method: "POST",
-        body: await f.arrayBuffer(),
+    if (f.size > warnFileSize) {
+      for (var start = 0; start < f.size; start += fileChunkSize) {
+        let end = Math.min(start + fileChunkSize, f.size);
+        const result = await fetch(
+          `${$settings.backendUrl}/root_resource_chunk?name=${f.name}&addr=${start}`,
+          {
+            method: "POST",
+            body: await f.slice(start, end),
+          }
+        );
```

Comment:
If you're feeling frisky, you can make this loop concurrent by turning it into an array with a `.map` call. Then you can collect the promises by doing something like `await Promise.all(startOffsets.map(i => doFetch(i)))`. 

Not sure if there will be much gain from many concurrent connections. Perhaps worth testing if you want? Not worth blocking the PR over this, though.

Reply:
Need to use .map more, i'll do that.